src/Emogrifier.php

```diff
@@ -670,42 +675,60 @@ private function parseCssRules($css)
     {
         $cssKey = md5($css);
         if (!isset($this->caches[static::CACHE_KEY_CSS][$cssKey])) {
-            // process the CSS file for selectors and definitions
-            preg_match_all('/(?:^|[\\s^{}]*)([^{]+){([^}]*)}/mi', $css, $matches, PREG_SET_ORDER);
-
-            $cssRules = [];
-            /** @var string[][] $matches */
-            /** @var string[] $cssRule */
-            foreach ($matches as $key => $cssRule) {
-                $cssDeclaration = trim($cssRule[2]);
-                if ($cssDeclaration === '') {
-                    continue;
-                }
-
-                $selectors = explode(',', $cssRule[1]);
-                foreach ($selectors as $selector) {
-                    // don't process pseudo-elements and behavioral (dynamic) pseudo-classes;
-                    // only allow structural pseudo-classes
-                    $hasPseudoElement = strpos($selector, '::') !== false;
-                    $hasAnyPseudoClass = (bool)preg_match('/:[a-zA-Z]/', $selector);
-                    $hasSupportedPseudoClass = (bool)preg_match(
-                        '/:(\\S+\\-(child|type\\()|not\\([[:ascii:]]*\\))/i',
-                        $selector
-                    );
-                    if ($hasPseudoElement || ($hasAnyPseudoClass && !$hasSupportedPseudoClass)) {
+            $splitCss = $this->splitCssAndMediaQuery($css);
+
+            $cssRules = [
+                'inlineable' => [],
+                'uninlineable' => []
```

Comment:
Please add a trailing comma also after the last element of a multiline array. This makes appending elements possible without having to modify the existing line.

Reply:
Done.